src/bindings/python/flux/accounting/job_archive_interface.py

```diff
@@ -444,7 +444,7 @@ def calc_usage_factor(
     )
     acct_conn.commit()
 
-    return usage_user_historical
+    return round(usage_user_historical, 1)
```

Comment:
Do you know which part of the test is non-deterministic (or maybe time-dependent) which could be causing the fluctuation at smaller decimal values?

Reply:
I _think_ the fluctuation could be here, where the historical job usage factor has a decay factor applied to it:

```python
usage_factor * math.pow(decay_factor, power)
```

Do you think it would be better to apply the rounding here?

Reply:
Ah, okay. After some more debugging I think I found it.

When a per-job factor is calculated, it takes two inputs: `job.nnodes` and `job.elapsed`:

```python
for job in job_totals_user:
    per_job_factors.append(round((job.nnodes * job.elapsed), 5))
```

Most of the tests use `mock.patch()` to simulate a period of time, which means this round to the 5th decimal place shouldn't affect a factor's value. However, the two tests that were consistently failing were required to use `time.time()` inside of `mock.patch()` to simulate a half-period further, thus effecting the value of `job.elapsed`, which in turn effects the historical usage factor. 

So I think this is what is causing the fluctuation. Maybe instead of rounding the final value and affecting `calc_usage_factor()`, I can instead round the `test` value in the unit test itself?

```python
self.assertEqual(round(usage_factor, 1), 4519.0)
```

Reply:
Awesome sleuthing @cmoussa1!

> So I think this is what is causing the fluctuation. Maybe instead of rounding the final value and affecting calc_usage_factor(), I can instead round the test value in the unit test itself?

That definitely seems like an improvement.

> However, the two tests that were consistently failing were required to use time.time() inside of mock.patch() to simulate a half-period further, thus effecting the value of job.elapsed, which in turn effects the historical usage factor.

Is it possible to use the mock patch to deterministically simulate the movement of time into the next half-period?  For example, maybe the mock return value for `time.time` can be changed within the test using [`Mock.return_value`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.return_value) or by passing a list of times to [`Mock.side_effect`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect).

Reply:
> Is it possible to use the mock patch to deterministically simulate the movement of time into the next half-period?

Ah, that's a good point - yes I can. At first I thought no, because the time period of the current half-life period is set when the database is initially created using `time.time()`, but I forgot to realize I can also simulate that as well in the unit test file with an update statement:

```python
# simulate end of half life period in FluxAccounting database
update_stmt = """
    UPDATE t_half_life_period_table SET end_half_life_period=?
    WHERE cluster='cluster'
    """
    acct_conn.execute(update_stmt, ("10000000",))
    acct_conn.commit()
```

Now, I think that none of the unit tests in **test_job_archive_interface.py** should be dependent on `time.time()`. I'll go ahead and push up a commit that integrates this change. If it looks good to you, I'll go ahead and squash this one and the previous commit.